### PR TITLE
Heading API changes

### PR DESCRIPTION
**Pull request by hadasfa**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +34,9 @@
       ],
       "testIds": [
         "./dist/esm/tests/test-ids-utils.d.ts"
+      ],
+      "next": [
+        "./dist/esm/next/next.d.ts"
```
Comment (hadasfa) [inline package.json:39]: d.ts file for next entry (needed for eslint to not throw error but not helps with autocomplete)

File: rollup.config.js [hunk 2]
```diff
@@ -44,6 +44,43 @@ function generateCssModulesMockName(name) {
...
+      // Move JS and DTS files
+      if (fs.existsSync(originalJsPath)) {
+        fs.renameSync(originalJsPath, targetJsPath);
+      }
+      if (fs.existsSync(originalDtsPath)) {
+        fs.renameSync(originalDtsPath, targetDtsPath);
+      }
+
+      // Create package.json in the 'next' folder
+      const packageData = {
```
Comment (hadasfa) [inline rollup.config.js:73]: tried it as a soultion for types (for autocomplete but not worked, will remove).
the rest of the plugin is for moving the d.ts and js file to one location which compiled by tsconfig (originally next.js was under esm which was problematic)

File: rollup.config.js [hunk 3]
```diff
@@ -56,11 +93,13 @@ export default {
     index: path.join(SRC_PATH, "index.js"),
     icons: path.join(SRC_PATH, "components/Icon/Icons/index.ts"),
     interactionsTests: path.join(SRC_PATH, "tests/interactions-utils.ts"),
-    testIds: path.join(SRC_PATH, "tests/test-ids-utils.ts")
+    testIds: path.join(SRC_PATH, "tests/test-ids-utils.ts"),
+    next: path.join(SRC_PATH, "next.ts")
```
Comment (hadasfa) [inline rollup.config.js:97]: from where to take the next.ts (next to index.js)

File: package.json [hunk 4]
```diff
@@ -18,6 +18,7 @@
     "./interactionsTests": "./dist/esm/src/tests/interactionsTests.js",
     "./testIds": "./dist/esm/src/tests/testIds.js",
     "./storybookComponents": "./dist/storybook/index.js",
+    "./next": "./dist/esm/next/next.js",
```
[the remark this answers is not in the document]
Comment (shlomitc) [reply package.json:21]: shouldn't it be  `"./next": "./dist/esm/src/next/next.js",`

**Review by shlomitc — APPROVED** [review]

**Comment by shlomitc** [comment]
just notice the chromatic UI changes

**Pull request merged**